lib/reconcile/vernemq.go

```diff
@@ -237,6 +237,38 @@ func getVerneMQEnvVars(statefulSetName string, cr *apiv1alpha1.Astarte) []v1.Env
 					Value: strconv.Itoa(cr.Spec.VerneMQ.DeviceHeartbeatSeconds * 1000),
 				})
 		}
+
+		if pointy.BoolValue(cr.Spec.VerneMQ.SSLListener, false && cr.Spec.VerneMQ.SSLListenerSecretRef != nil &&
+			cr.Spec.VerneMQ.SSLListenerSecretRef.Name != "" &&
+			cr.Spec.VerneMQ.SSLListenerSecretRef.Namespace != "") {
```

Comment:
Is namespace information compulsory in the ref? I don't recall if namespace is always needed or, if blank, it matches the namespace of the parent resource. Depending on this information you might want to revise the need for having the field specified

Reply:
It is an optional field. However the [doc](https://pkg.go.dev/k8s.io/api/core/v1#SecretReference) states that `Namespace defines the space within which the secret name must be unique`. Since the behavior is not clearly defined when the namespace is not set, I would keep the checks as they are.